### ISSUE TITLE
False negative when passing protocol class as argument to a constructor

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7475]
### Discussed in https://github.com/microsoft/pyright/discussions/7465

The following code should produce an error but does not.

```python
from typing import Generic, Protocol, TypeVar
from abc import ABC

T = TypeVar("T")

class Injectable(Generic[T]):
    def __new__(cls, i: type[T]) -> T:
        return i()

class IMailService_Proto(Protocol): ...

class IMailService_ABC(ABC): ...

m1 = Injectable(IMailService_Proto)  # Mypy produces an error here, as it should
reveal_type(m1)

m2 = Injectable(IMailService_ABC)
reveal_type(m2)
```

**Comment by erictraut** [comment]
This gets into a gray area of the typing spec. I've started a discussion in the [Python typing forum](https://discuss.python.org/t/compatibility-of-protocol-class-object-with-type-t-and-type-any/48442) to get feedback from the community.

**Comment by erictraut** [comment]
There is not yet consensus on this topic, so I'm going to close for now. I'll revisit this if/when this topic is clarified in the typing spec.
